### ISSUE TITLE
Windows Defender reports Trojan:Win32/Emali.A!cl

### ISSUE THREAD

**Issue by Ponjimon** [lian/msfs2020-go#13]

**Comment by jpeg** [comment]
Defender didn't detect it on my PC with the latest definition updates. However, VirusTotal shows MaxSecure detecting it as a threat via what appears to be a generic rule `Trojan.Malware.300983.susgen` and Cylance as `Unsafe`: https://www.virustotal.com/gui/file/b5b5eae304f9cab0c29f5b82302468e6d7ab197ef1994bc882652c172fbc99af/detection

Embedding and extracting a DLL from inside the main executable would seem to be the most likely culprit to trip behavior based detection rules: https://github.com/lian/msfs2020-go/blob/v0.0.5/simconnect/bindata.go#L71. Is there a reason it can't be simply be included in the release ZIP alongside the executable?

**Comment by lian** [comment]
this is a false postive.

@jpeg yea, please see here https://www.reddit.com/r/MicrosoftFlightSim/comments/ih5m57/vfrmap_is_it_malware/

i thought the same, that embedding and extracting the dll makes MaxSecure unhappy. and tried a build without that embedding, but MaxSecure still didn't like it. :(

will do some more tests. maybe Cylance removes their warning without embedding.

to why i embeded: i wanted to have just one single exe, so the user can't mess up putting the simconnect.dll file to the correct place.

**Comment by lian** [comment]
@jpeg funny enough, the `Trojan:Win32/Emali.A!cl` reported by microsoft scanner of virustotal went away when i rebuild the exact same v0.0.5 code. golang just created a new exe of the exact same code. but the MaxSecure and Cylance seem to not go away.

**Comment by jpeg** [comment]
Did a quick and dirty test by removing https://github.com/lian/msfs2020-go/blob/v0.0.5/simconnect/simconnect.go#L43-L49 & `simconnect/bindata.go`. Both files came up clean:
https://www.virustotal.com/gui/file/9d4d9b0306884de05bf4c5562be5b15bcc3dc44dc3111e362b220eae744f2a89/detection
https://www.virustotal.com/gui/file/677fd3849d54e2a6831dc1aa348d974ead2255ec609fbf5830ef32bf0804b8bc/detection

**Comment by lian** [comment]
@jpeg nice, let me try again. i tried an hour ago and MaxSecure still didn't like it. will try again. i also uploaded a new v0.0.5 (just a rebuild of the exact same code, and defender likes it and doesn't think its `Trojan:Win32/Emali.A!cl` anymore.

**Comment by lian** [comment]
@jpeg this is madness, if i compile it with

`GOOS=windows GOARCH=amd64 go build -ldflags "-X main.buildVersion=$build_version -X main.buildTime=$build_time" -v github.com/lian/msfs2020-go/vfrmap`

what my `build-vfrmap.sh` does

then MaxSecure thinks its `Trojan.Malware.300983.susgen`.

the ldflags triggers it

**Comment by jpeg** [comment]
Been trying various tweaks and building with the script (note: using bash on Windows). No matter what I change, about one in two builds trip that Max Secure Trojan rule with no discernable pattern... I have even literally had two builds with identical code where one detected and the other did not.

Only theory I have remaining is that it is indeed a golang related false positive as you suggested on Reddit. Max Secure has a false positive submission form you could try if you'd like: https://www.maxsecureantivirus.com/submit_aFalse_Positive.htm

That said, Max Secure doesn't seem that common so probably not a huge deal? Appears to be an Indian company so possible they just aren't turning up much in English search results, but probably a small slice of the MSFS userbase either way.

**Comment by lian** [comment]
same here, been trying various ways. currently bang my head again a zip failing MaxSecure while the exe alone passes. it really just the zip container that makes it fail.

good point on potentially just ignoring MaxSecure. but given the reddit thread i would love to post fully passing virustotal links with any future release. but given 3 hours wasted so far i'm almost ready to give up for now. unfortunate having to defend your open source code against such false positive scans.

**Comment by lian** [comment]
closing this issue and moving the MaxSecure false-postive debugging [here](https://github.com/lian/msfs2020-go/issues/14)

`Trojan:Win32/Emali.A!cl` false-postive from this issue is solved in v0.0.5

https://www.virustotal.com/gui/file/946f9fa093369c29e3f8db51293b84bc7296ae4ed68112b2aebfa71ae85e349c/detection
